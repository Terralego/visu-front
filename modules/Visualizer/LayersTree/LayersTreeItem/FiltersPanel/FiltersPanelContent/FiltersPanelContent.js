import React from 'react';
import classnames from 'classnames';
import Filters, { TYPE_RANGE } from '../../../../../Forms/Filters';

const formatRange = ({ min, max, value, ...property }) => ({
  ...property,
  labelStepSize: max || 100,
  value: value || [min || 0, max || 100],
  min: min || 0,
  max: max || 100,
});

export class FiltersPanelContent extends React.Component {
  static defaultProps = {
    onRef () {},
  };

  state = {};

  elRef = React.createRef();

  componentDidMount () {
    const { onRef } = this.props;
    onRef(this.elRef.current);
  }

  componentDidUpdate ({ visible: prevVisible }) {
    const { visible } = this.props;

    if (visible !== prevVisible) {
      this.animate();
    }
  }

  animate () {
    const { visible } = this.props;

    if (visible) {
      this.setState({
        animate: true,
      });
      setTimeout(() => this.setState({
        visible: true,
      }));
    } else {
      this.setState({
        visible: false,
      });
      setTimeout(() => this.setState({
        animate: false,
      }), 150);
    }
  }

  render () {
    const {
      top,
      left,
      width,
      onChange,
      layer: { filters: { form = [] } = {} },
      filtersValues,
    } = this.props;
    const { visible, animate } = this.state;
    const { elRef } = this;

    const filters = form
      .map(filter => (filter.type === TYPE_RANGE
        ? formatRange(filter)
        : filter));

    return (
      <div
        ref={elRef}
        className={classnames('filters-panel', {
          'filters-panel--visible': visible,
          'filters-panel--animate': animate,
          'bp3-dark': true,
        })}
        style={{
          left: left && width && left + width,
        }}
      >
        <div
          className="filters-panel__arrow"
          style={{
            top,
          }}
        />
        <div className="filters-panel__content">
          <div
            className="filters-panel__header"
          >
            <h2 className="filters-panel__title">Filtrer</h2>
          </div>
          <div
            className="filters-panel__form"
          >
            <Filters
              onChange={onChange}
              filters={filters}
              properties={filtersValues}
              locales={{
                emptySelectItem: 'Toutes',
                noResults: 'Aucun rÃ©sultat',
                overlappingDatesMessage: 'Date chevauchante',
                invalidDateMessage: 'Date invalide',
                startInputProps: 'Du JJ/MM/AAAA',
                endInputProps: 'Au JJ/MM/AAAA',
              }}
            />
          </div>
        </div>
      </div>
    );
  }
}

export default FiltersPanelContent;
