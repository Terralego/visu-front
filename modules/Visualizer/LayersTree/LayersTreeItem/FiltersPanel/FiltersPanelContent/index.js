import FiltersPanelContent from './FiltersPanelContent';

import { connectLayersTree } from '../../../LayersTreeProvider/context';

export default connectLayersTree('translate', ({ getLayerState }, { layer, layer: { exclusive, layers } }) => {
  const activeLayer = exclusive
    ? layers.find(l => getLayerState({ layer: l }).active) || layer
    : layer;

  return {
    filtersValues: getLayerState({ layer: activeLayer }).filters || {},
    activeLayer,
  };
})(FiltersPanelContent);
