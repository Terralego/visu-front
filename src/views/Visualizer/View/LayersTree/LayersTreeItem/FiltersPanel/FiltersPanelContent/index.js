import FiltersPanelContent from './FiltersPanelContent';

import { connectView } from '../../../../context';

export default connectView(({ getLayerState }, { layer }) => ({
  filtersValues: getLayerState({ layer }).filters || {},
}))(FiltersPanelContent);
