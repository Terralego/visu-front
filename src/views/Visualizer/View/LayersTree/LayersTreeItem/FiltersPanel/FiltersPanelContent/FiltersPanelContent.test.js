import React from 'react';
import renderer from 'react-test-renderer';
import FiltersPanelContent from './FiltersPanelContent';


it('should render correctly', () => {
  const tree = renderer.create((
    <FiltersPanelContent layer={{ filters: { form: [] } }} />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
