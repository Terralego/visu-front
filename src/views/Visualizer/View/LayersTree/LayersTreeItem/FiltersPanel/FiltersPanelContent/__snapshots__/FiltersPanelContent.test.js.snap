// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="filters-panel bp3-dark"
  style={
    Object {
      "left": undefined,
    }
  }
>
  <div
    className="filters-panel__arrow"
    style={
      Object {
        "top": undefined,
      }
    }
  />
  <div
    className="filters-panel__content"
  >
    <div
      className="filters-panel__header"
    >
      <h2
        className="filters-panel__title"
      >
        Filtrer
      </h2>
    </div>
    <div
      className="filters-panel__form"
    >
      <div />
    </div>
  </div>
</div>
`;
